### DISCUSSION TITLE
how to Install off-line packages

### DISCUSSION THREAD

**Discussion by sunlight905** [deepmodeling/deepmd-kit#2969] [Q&A] [answered]
how to use the file "deepmd-kit-2.2.7-cpu-Linux-x86_64.sh" for installing the deepmd-kit?
is it the commond "sudo bash deepmd-kit-2.2.7-cpu-Linux-x86_64.sh"

**Comment by njzjz** [comment]
Use `sudo` only when you need to install it into a directory that you don't have the write access.

**Reply by sunlight905** [reply]
Thanks for your reply.
But when I use the command, there are some errors below, and I have download the packages twice from github, and the errors always exists.
Do you accept the license terms? [yes|no]
>>> yes

deepmd-kit will now be installed into this location:
/home/sunlight/deepmd-kit

  - Press ENTER to confirm the location
  - Press CTRL-C to abort the installation
  - Or specify a different location below

[/home/sunlight/deepmd-kit] >>> /home/sunlight/work2/dpmd/2.2.7
PREFIX=/home/sunlight/work2/dpmd/2.2.7
WARNING: md5sum mismatch of tar archive
expected: 589d947347811a21b1044490c8646470
     got: 3091fd4ccd0b72fbc7846118e5fea194  -
Unpacking payload ...
./deepmd-kit-2.2.7-cpu-Linux-x86_64.sh：line 434: 75263 completed(141)          extract_range $boundary1 $boundary2
     75264 Segment error (core has been dumped)| "$CONDA_EXEC" constructor --extract-tarball --prefix "$PREFIX"

**Reply by njzjz** [reply]
I tried to install it in a new Google Colab environment, and nothing went wrong: https://gist.github.com/njzjz/e4a356c52d1ca4b389c5117507b01db4

The md5sum mismatch should not be caused by the file itself.

**Reply by sunlight905** [reply, marked as answer]
I download the packages in the windows system. and then transfer it to linux, and the error occurs.
The reason of the error is:
A text file under Windows uploaded to Unix/Linux using ssh will be automatically formatted, that is, convert carriage returns and line feeds to line feeds, and when you view it under Unix, you will see that it is ^M less.

and the solution is Compress the file into a zip file and then transfer it, unzip it into sh after successful transfer and it will be successfully installed.

Thanks for your reply!
